packages/internal-tooling/babel/index.js

```diff
@@ -1,18 +1,23 @@
-const isBabelRegister = caller => {
-  return !!(caller && caller.name === '@babel/register')
+const isNodeCaller = caller => {
+  return caller && (caller.name === '@babel/register' || caller.name === 'babel-jest')
 }
-const isJest = caller => {
-  return !!(caller && caller.name === 'babel-jest')
+const isDistCaller = caller => {
+  return !!(caller && caller.name === 'babel-gulp')
+}
+const supportsESM = caller => {
+  return !!((caller && caller.name === 'babel-loader') || caller.useESModules)
 }
 
 module.exports = api => {
-  const isNode = api.caller(isBabelRegister) || api.caller(isJest)
+  const isDistBuild = api.caller(isDistCaller)
```

Comment:
nit: probably, `isDistBundle` name will better reflect semantics and related optimizations applied for this scenario